### PR TITLE
Adding information on sigstore-go

### PR DESCRIPTION
**Pull request by hayleycd**
#### Summary
Partially addresses #324 by adding information on `sigstore-go`

#### Release Note
None

### CODE REVIEW

**Comment by hayleycd** [comment]
@codysoyland @steiza tagging you here in case you want to review. Otherwise, @haydentherapper or @cmurphy can you take a look?

File: content/en/language_clients/go/overview.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+---
+type: docs
+category: Go
+title: Go Client Overview
+weight: 5
+---
+
+[`sigstore-go`](https://pkg.go.dev/github.com/sigstore/sigstore-go) is the Go language client for Sigstore.
+
+[Cosign](../../cosign/signing/overview.md) is a command line utility that is used to sign software artifacts and verify signatures using Sigstore and is also written in Go. `sigstore-go` is a lighter weight tool and boasts the following benefits:
```
Comment (steiza) [inline content/en/language_clients/go/overview.md:10]:
```suggestion
[`sigstore-go`](https://pkg.go.dev/github.com/sigstore/sigstore-go) is the Go language client library for Sigstore.

`sigstore-go` is intended as a minimal dependency library for signing and verifying. It's not intended to replace [cosign](../../cosign/signing/overview.md), which provides a CLI with many features for interacting with Sigstore. Over time, `cosign` will use `sigstore-go` for verification.
```

**Review by steiza — APPROVED** [review]
One minor suggestion to clarify the difference between sigstore-go and cosign, otherwise looks good to me!

**Comment by hayleycd** [comment]
Thank you for the review @steiza. I made the suggested change (and had to fix the DCO) but we should be good now.

**Review by haydentherapper — APPROVED** [review]

**Pull request merged**